br/tests/utils.go

```diff
@@ -0,0 +1,207 @@
+// Copyright 2024 PingCAP, Inc. Licensed under Apache-2.0.
+
+package main
+
+import (
+	"encoding/binary"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/klauspost/compress/zstd"
+)
+
+const (
+	cmdValidateBackupFiles = "validateBackupFiles"
+	extSST                 = ".sst"
+	extLOG                 = ".log"
+)
+
+func main() {
+	if len(os.Args) < 2 {
+		fmt.Println("Usage: go run utils.go <command> [arguments]")
+		fmt.Println("Available commands:")
+		fmt.Println("  validateBackupFiles")
+		os.Exit(1)
+	}
+
+	switch os.Args[1] {
+	case cmdValidateBackupFiles:
+		if !validateBackupFiles(os.Args[2:]) {
+			fmt.Println("validation failed")
+			os.Exit(1)
+		}
+	default:
+		fmt.Printf("Unknown command: %s\n", os.Args[1])
+		os.Exit(1)
+	}
+}
+
+func validateBackupFiles(args []string) bool {
+	validateCmd := flag.NewFlagSet(cmdValidateBackupFiles, flag.ExitOnError)
+	cmd := validateCmd.String("command", "", "Backup or restore command")
+	encryptionArg := validateCmd.String("encryption", "", "Encryption argument")
+
+	err := validateCmd.Parse(args)
+	if err != nil {
+		fmt.Println("Failed to parse arguments")
+		return false
+	}
+
+	if *cmd == "" {
+		fmt.Println("Please provide the full backup or restore command using --command flag")
+		validateCmd.PrintDefaults()
+		return false
+	}
+
+	storagePath, found := parseCommand(*cmd)
+	// doesn't need to validate if it's not doing backup/restore
+	if !found {
+		fmt.Println("No need to validate")
+		return true
+	}
+
+	fmt.Printf("Validating files in: %s\n", storagePath)
+	return checkCompressionAndEncryption(storagePath, *encryptionArg)
+}
+
+func parseCommand(cmd string) (string, bool) {
+	args := strings.Fields(cmd)
+	hasBackupOrRestore := false
+	storagePath := ""
+
+	for i, arg := range args {
+		if arg == "backup" || arg == "restore" {
```

Comment:
Perhaps verifying backup is enough? As restore in fact contributes nothing to the files to be checked.

Reply:
I should add a comment for the code here.
For the full backup we can check the storage encryption status once the command is complete since it's a blocking call, but for log backup I believe we need some extra logic to wait for it to generate some files. So my approach is to verify log backup files before doing restore, making sure the files are actually encrypted. Let me know what you think!

Reply:
LGTM